File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-0.cpp
class KthLargest {
public:
    int kth=0;
    priority_queue<int>q;
    KthLargest(int k, vector<int>& nums) {
        kth=k;
        for(int num:nums){
            q.push(num);
        }
    }
    
    int add(int val) {
        q.push(val);
        stack<int>st;
        for(int i=0;i<kth-1;i++){
           st.push(q.top());
           q.pop();
        }

        int ans=q.top();
        
        while(!st.empty()){
            q.push(st.top());
            st.pop();
        }

        return ans;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {


        int n=temperatures.length;
        int[] ans= new int[n];

        Stack<Integer>st=new Stack<>();
        int j=n-1;

        while(j>=0){
            while(!st.empty() && temperatures[j]>=temperatures[st.peek()]){
                int top=st.pop();
                 if(st.empty()) ans[top]=0;
                 else ans[top]=st.peek()-top;
                
            }

            st.add(j);
            j--;
        }

        while(!st.empty()){
            int top=st.pop();
                 if(st.empty()) ans[top]=0;
                 else ans[top]=st.peek()-top;
        }

        return ans;
    }
}
